sources/web/datalab/polymer/components/file-browser/file-browser.ts

```diff
@@ -323,6 +336,9 @@ class FileBrowserElement extends Polymer.Element implements DatalabPageElement {
       const btn = document.createElement('paper-button');
       btn.classList.add('toolbar-button');
       btn.addEventListener('click', () => {
+        if (this._fetching) {
+          return;
```

Comment:
So if the page happens to be fetching when I click the switcher button, it just does nothing? This seems like a poor UX.

Reply:
Yes, it's an easy fix for the current state, which is inconsistent.
The other, more involved, option would be to queue all those requests and play them one by one,  but I'm not sure that's a great experience either, because there's no easy way to interrupt that queue.

Reply:
OK, please add a TODO.

Reply:
Done.